### DISCUSSION TITLE
mongodb sink

### DISCUSSION THREAD

**Discussion by rkedward** [vectordotdev/vector#13976] [General]
Hi, is there a mongodb sink for vector?  I'd like to use vector to send events to a mongodb collection.  If not, can I extend vector by writing my own sink as a plugin?

**Comment by jszwedko** [comment]
Hi @rkedward ! There is not one. Would you want to open an issue? There isn't a plugin system yet, but we would be happy to have a contribution to Vector itself of a MongoDB sink. Otherwise, users have approximated a plugin system by writing a separate process that exposes an HTTP endpoint and using the `http` sink from Vector to send data.

**Comment by rkedward** [comment]
@jszwedko, thanks for the quick response.  I'll file an issue.  It would be ideal if vector had a plugin system to allow users to create custom source/sinks/transforms.  I noticed a that some streaming processor projects are using wasm modules for this (redpanda, fluvio).

**Reply by jszwedko** [reply]
Thanks @rkedward ! We are also considering using WASM to support plugins in the future.
